<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tess and Tural" />


<title>IMC data analysis workflow</title>

<script src="site_libs/header-attrs-2.20/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">221201_workflowr_haibin</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Contents</a>
</li>
<li>
  <a href="0_prep_data.html">Data prep</a>
</li>
<li>
  <a href="1_QC_mask.html">QC mask</a>
</li>
<li>
  <a href="2_IMCA_cell_annotation.html">Cell ann</a>
</li>
<li>
  <a href="3_dim_red.html">Dim red</a>
</li>
<li>
  <a href="4_img_vis.html">Img vis</a>
</li>
<li>
  <a href="5_spatial_vis.html">Spatial</a>
</li>
<li>
  <a href="6_statistics.html">Stats</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">IMC data analysis workflow</h1>
<h4 class="author">Tess and Tural</h4>
<h4 class="date">University of Bern, DBMR, Visceral and Transplantation
Surgery lab</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-03-21
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>221201_workflowr_haibin/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221201code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221201)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221201code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221201)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
cell coverage
</li>
<li>
cell size
</li>
<li>
cells per mm2
</li>
<li>
compare UMAP and TSNE
</li>
<li>
display images
</li>
<li>
heatmap qc
</li>
<li>
load-libs
</li>
<li>
load data
</li>
<li>
mean marker expression across samples
</li>
<li>
open rds
</li>
<li>
read in data
</li>
<li>
reading in images
</li>
<li>
save spe1
</li>
<li>
segmentation QC
</li>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
single cell expression heatmap
</li>
<li>
SNR cell level
</li>
<li>
staining QC
</li>
<li>
UMAP and TSNE
</li>
<li>
UMAP target overlay
</li>
<li>
unnamed-chunk-1
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory
<code>1_QC_mask_cache</code> and re-run the analysis. To have workflowr
automatically delete the cache directory prior to building the file, set
<code>delete_cache = TRUE</code> when running <code>wflow_build()</code>
or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
C:/Users/IMC Platform
laptop/Users2/Tess/SteinbockCellSegmentation/221130_haibin/221201_workflowr_haibin
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong75a88a8">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong> 75a88a8
</a>
</p>
</div>
<div id="strongRepositoryversionstrong75a88a8"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version 75a88a8.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/0_prep_data_cache/
    Ignored:    analysis/1_QC_mask_cache/
    Ignored:    analysis/2_IMCA_cell_annotation_cache/
    Ignored:    analysis/3_dim_red_cache/
    Ignored:    analysis/4_img_vis_cache/
    Ignored:    analysis/5_spatial_vis_cache/
    Ignored:    analysis/6_statistics_cache/

Untracked files:
    Untracked:  Umap_target_overlay.pdf
    Untracked:  Umap_target_overlay.png
    Untracked:  allchannels_multidimplot.pdf
    Untracked:  data/images.csv
    Untracked:  data/images.rds
    Untracked:  data/img/
    Untracked:  data/intensities/
    Untracked:  data/masks.rds
    Untracked:  data/masks/
    Untracked:  data/neighbors/
    Untracked:  data/panel.csv
    Untracked:  data/raw/
    Untracked:  data/regionprops/
    Untracked:  data/sample_metadata.xlsx
    Untracked:  data/sample_metadata_shortened.xlsx
    Untracked:  output/lung_rffit.rds
    Untracked:  output/lung_rffit_v2.rds
    Untracked:  output/spe0.rds
    Untracked:  output/spe1.rds
    Untracked:  output/spe1_imca.rds
    Untracked:  output/spe3.rds
    Untracked:  output/spe4.rds
    Untracked:  output/spe5.rds
    Untracked:  output/spe_ca.rds
    Untracked:  output/spe_ca_v2.rds
    Untracked:  output/spe_ca_v2_dr.rds
    Untracked:  umap_metadata.png
    Untracked:  umap_target.png

Unstaged changes:
    Modified:   analysis/_site.yml

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/1_QC_mask.Rmd</code>) and HTML
(<code>docs/1_QC_mask.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
75a88a8
</td>
<td>
tessbrodie
</td>
<td>
2023-03-21
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="quality-control-cell-mask" class="section level1">
<h1>Quality control cell mask</h1>
<style type="text/css">
.scroll-100 {
  max-height: 100px;
  overflow-y: auto;
  background-color: inherit;
}
</style>
<div id="load-packages" class="section level3">
<h3>Load packages</h3>
<pre class="r"><code>library(imcRtools)
library(cytomapper)
library(openxlsx)
library(stringr)
library(dittoSeq)
library(RColorBrewer)
library(cytomapper)
library(viridis)
library(tidyverse)
library(ggrepel)
library(EBImage)
library(scuttle)
library(mclust)
library(scater)
library(patchwork)
library(batchelor)
library(harmony)
library(Seurat)
library(SeuratObject)
library(cowplot)
library(gtools)</code></pre>
</div>
<div id="load-data" class="section level3">
<h3>Load data</h3>
<pre class="r"><code>spe &lt;- readRDS(file.path(&quot;output&quot;, &quot;spe0.rds&quot;))
spe$ROI &lt;- factor(spe$ROI, levels = mixedsort(unique(spe$ROI)))
spe$patient_id &lt;- factor(spe$patient_id, levels = 1:9)
use_channel &lt;- rowData(spe)$name[rowData(spe)$use_channel]</code></pre>
</div>
<div id="read-in-images-and-create-metadata-for-images-object"
class="section level3">
<h3>Read in images and create metadata for images object</h3>
</div>
<div id="save-rds" class="section level3">
<h3>Save rds</h3>
</div>
<div id="open-rds" class="section level3">
<h3>Open rds</h3>
<pre class="r"><code>images &lt;- readRDS(&quot;data/images.rds&quot;)
masks &lt;- readRDS(&quot;data/masks.rds&quot;)</code></pre>
</div>
<div id="segmentation-quality-control" class="section level3">
<h3>Segmentation quality control</h3>
<pre class="r"><code>img_ids &lt;- sample(seq_len(length(images)), 6)

# Normalize and clip images
cur_images &lt;- images[img_ids]
cur_images &lt;- normalize(cur_images, separateImages = TRUE)
cur_images &lt;- normalize(cur_images, inputRange = c(0,0.2))


plotPixels(cur_images,
           mask = masks[img_ids],
           img_id = &quot;sample_id&quot;,
           missing_colour = &quot;white&quot;,
           colour_by = c(&quot;CD206&quot;, &quot;E-cadherin&quot;, &quot;DNA1&quot;, &quot;DNA2&quot;, &quot;HistoneH3&quot;),
           colour = list(&quot;CD206&quot; = c(&quot;black&quot;, &quot;green&quot;),
                         &quot;E-cadherin&quot; = c(&quot;black&quot;, &quot;red&quot;),
                         &quot;DNA1&quot; = c(&quot;black&quot;, &quot;blue&quot;), &quot;DNA2&quot;= c(&quot;black&quot;, &quot;blue&quot;),
                         &quot;HistoneH3&quot; = c(&quot;black&quot;, &quot;blue&quot;)),
           bcg = list(&quot;DNA1&quot; = c(0, 1, 1),
                      &quot;DNA2&quot; = c(0, 1, 1),
                      &quot;HistoneH3&quot; = c(0, 1, 1),
                      &quot;E-cadherin&quot; = c(0, 3, 1),
                      &quot;CD206&quot; = c(0, 5, 1)),
           image_title = NULL,
           legend = list(colour_by.title.cex = 0.7,
                         colour_by.labels.cex = 0.7))</code></pre>
<p><img src="figure/1_QC_mask.Rmd/segmentation QC-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="visualize-single-cell-expression-in-a-heatmap"
class="section level3">
<h3>Visualize single-cell expression in a heatmap</h3>
<pre class="r"><code>set.seed(20221201)
cur_cells &lt;- sample(seq_len(ncol(spe)), 2000)

dittoHeatmap(spe[,cur_cells], genes = use_channel,
             assay = &quot;exprs&quot;, cluster_cols = TRUE, scale = &quot;none&quot;,
             heatmap.colors = viridis(100), annot.by = &quot;condition&quot;,
             annotation_colors = list(indication = metadata(spe)$color_vectors$indication))</code></pre>
<p><img src="figure/1_QC_mask.Rmd/single cell expression heatmap-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="image-level-qc-signal-to-noise-ratio-snr-for-each-marker"
class="section level3">
<h3>Image level QC: signal to noise ratio (SNR) for each marker</h3>
<pre class="r"><code>img_ids &lt;- sample(seq_len(length(images)), 25)

cur_snr &lt;- lapply(images[img_ids], function(img){
    mat &lt;- apply(img, 3, function(ch){
        # Otsu threshold
        thres &lt;- otsu(ch, range = c(min(ch), max(ch)))
        # Signal-to-noise ratio
        snr &lt;- mean(ch[ch &gt; thres]) / mean(ch[ch &lt;= thres])
        # Signal intensity
        ps &lt;- mean(ch[ch &gt; thres])
        
        return(c(snr = snr, ps = ps))
    })
    t(mat) %&gt;% as.data.frame() %&gt;% 
        mutate(marker = colnames(mat)) %&gt;% 
        pivot_longer(cols = c(snr, ps))
})

cur_snr &lt;- do.call(rbind, cur_snr)

#Subsetting the SNR plot values to only contain channels of interest

cur_snr &lt;- cur_snr[cur_snr$marker %in% 
          (rownames(spe)[rowData(spe)$use_channel]), ]

cur_snr %&gt;% 
    group_by(marker, name) %&gt;%
    summarize(mean = mean(value),
              ci = qnorm(0.975)*sd(value)/sqrt(n())) %&gt;%
    pivot_wider(names_from = name, values_from = c(mean, ci)) %&gt;%
    ggplot() +
    geom_errorbar(aes(y = log2(mean_snr), xmin = log2(mean_ps - ci_ps), 
                      xmax = log2(mean_ps + ci_ps))) +
    geom_errorbar(aes(x = log2(mean_ps), ymin = log2(mean_snr - ci_snr), 
                      ymax = log2(mean_snr + ci_snr))) +
    geom_point(aes(log2(mean_ps), log2(mean_snr))) +
    geom_label_repel(aes(log2(mean_ps), log2(mean_snr), label = marker)) +
    theme_minimal(base_size = 15) + ylab(&quot;Signal-to-noise ratio&quot;) +
    xlab(&quot;Signal intensity&quot;)</code></pre>
<pre><code>`summarise()` has grouped output by &#39;marker&#39;. You can override using the
`.groups` argument.</code></pre>
<pre><code>Warning in FUN(X[[i]], ...): NaNs produced

Warning in FUN(X[[i]], ...): NaNs produced</code></pre>
<p><img src="figure/1_QC_mask.Rmd/heatmap qc-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="image-area-covered-by-cells" class="section level3">
<h3>Image area covered by cells</h3>
<pre class="r"><code>colData(spe) %&gt;%
    as.data.frame() %&gt;%
    group_by(ROI) %&gt;%
    summarize(cell_area = sum(area),
           no_pixels = mean(width_px) * mean(height_px)) %&gt;%
    mutate(covered_area = cell_area / no_pixels) %&gt;%
    ggplot() +
        geom_point(aes(ROI, covered_area)) + 
        theme_minimal(base_size = 15) +
        ylim(c(0, 1)) + 
        theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) +
        ylab(&quot;% covered area&quot;) + xlab(&quot;&quot;)</code></pre>
<p><img src="figure/1_QC_mask.Rmd/cell coverage-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="display-single-images-with-low-image-area-covered-by-cells-to-analyze"
class="section level3">
<h3>Display single images with low image area covered by cells to
analyze</h3>
<pre class="r"><code># Normalize and clip images
cur_images &lt;- images[c(&quot;221031_day1_Haibin_6_11&quot;, &quot;221031_day1_Haibin_7_12&quot;, &quot;221103_day4_Haibin_54_18&quot;)]
cur_images &lt;- normalize(cur_images, separateImages = TRUE)
cur_images &lt;- normalize(cur_images, inputRange = c(0, 0.2))

plotPixels(cur_images,
           mask = masks[c(&quot;221031_day1_Haibin_6_11&quot;, &quot;221031_day1_Haibin_7_12&quot;, &quot;221103_day4_Haibin_54_18&quot;)],
           img_id = &quot;sample_id&quot;,
           missing_colour = &quot;white&quot;,
           colour_by = c(&quot;CD206&quot;, &quot;a-SMA&quot;, &quot;CD3e&quot;, &quot;E-cadherin&quot;, &quot;DNA1&quot;),
           colour = list(&quot;CD206&quot; = c(&quot;black&quot;, &quot;yellow&quot;),
                         &quot;a-SMA&quot; = c(&quot;black&quot;, &quot;red&quot;),
                         &quot;CD3e&quot; = c(&quot;black&quot;, &quot;green&quot;),
                         &quot;E-cadherin&quot; = c(&quot;black&quot;, &quot;cyan&quot;),
                         &quot;DNA1&quot; = c(&quot;black&quot;, &quot;blue&quot;)),
           legend = list(colour_by.title.cex = 0.7,
                         colour_by.labels.cex = 0.7))</code></pre>
<p><img src="figure/1_QC_mask.Rmd/display images-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div
id="mean-marker-expression-per-image-to-identify-images-with-outlying-cell-type-compositions"
class="section level3">
<h3>Mean marker expression per image to identify images with outlying
cell type compositions</h3>
<pre class="r"><code>image_mean &lt;- aggregateAcrossCells(spe, 
                                   ids = spe$ROI, 
                                   statistics=&quot;mean&quot;,
                                   use.assay.type = &quot;counts&quot;)

assay(image_mean, &quot;exprs&quot;) &lt;- asinh(counts(image_mean))

dittoHeatmap(image_mean, genes = use_channel,
             assay = &quot;exprs&quot;, cluster_cols = TRUE, scale = &quot;none&quot;,
             heatmap.colors = viridis(100), 
             annot.by = c(&quot;condition&quot;, &quot;patient_id&quot;, &quot;ROI&quot;),
             show_colnames = TRUE)</code></pre>
<p><img src="figure/1_QC_mask.Rmd/mean marker expression across samples-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="cell-level-qc-snr-for-each-marker" class="section level3">
<h3>Cell level QC: SNR for each marker</h3>
<pre class="r"><code>set.seed(20221201)
mat &lt;- apply(assay(spe, &quot;exprs&quot;), 1, function(x){
    cur_model &lt;- Mclust(x, G = 2)
    mean1 &lt;- mean(x[cur_model$classification == 1])
    mean2 &lt;- mean(x[cur_model$classification == 2])
    
    signal &lt;- ifelse(mean1 &gt; mean2, mean1, mean2)
    noise &lt;- ifelse(mean1 &gt; mean2, mean2, mean1)
    
    return(c(snr = signal/noise, ps = signal))
})
    
cur_snr &lt;- t(mat) %&gt;% as.data.frame() %&gt;% 
        mutate(marker = colnames(mat))

# Subsetting the SNR plot values to only contain channels of interest
cur_snr &lt;- cur_snr[cur_snr$marker %in% 
                     c(rownames(spe)[rowData(spe)$use_channel]), ]

cur_snr %&gt;% ggplot() +
    geom_point(aes(log2(ps), log2(snr))) +
    geom_label_repel(aes(log2(ps), log2(snr), label = marker)) +
    theme_minimal(base_size = 15) + ylab(&quot;Signal-to-noise ratio&quot;) +
    xlab(&quot;Signal intensity&quot;)</code></pre>
<p><img src="figure/1_QC_mask.Rmd/SNR cell level-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="distributions-of-cell-size-across-the-individual-images"
class="section level3">
<h3>Distributions of cell size across the individual images</h3>
<pre class="r"><code>colData(spe) %&gt;%
    as.data.frame() %&gt;%
    group_by(ROI) %&gt;%
    ggplot() +
        geom_boxplot(aes(ROI, area)) +
        theme_minimal(base_size = 15) + 
        theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) +
        ylab(&quot;Cell area&quot;) + xlab(&quot;&quot;)</code></pre>
<p><img src="figure/1_QC_mask.Rmd/cell size-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#table with average cell size and areas:
summary(spe$area)</code></pre>
<pre class="scroll-100"><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
    2.0    32.0    47.0    49.5    63.0   493.0 </code></pre>
<pre class="r"><code># remove very large cells from the dataset:
sum(spe$area &gt; 150) </code></pre>
<pre class="scroll-100"><code>[1] 1435</code></pre>
<pre class="r"><code># remove them from the dataset
spe &lt;- spe[, spe$area &lt;= 150]</code></pre>
</div>
<div id="percentage-of-image-area-covered-by-cells"
class="section level3">
<h3>Percentage of image area covered by cells</h3>
<pre class="r"><code>colData(spe) %&gt;%
    as.data.frame() %&gt;%
    group_by(ROI) %&gt;%
    summarize(cell_count = n(),
           no_pixels = mean(width_px) * mean(height_px)) %&gt;%
    mutate(cells_per_mm2 = cell_count/(no_pixels/1000000)) %&gt;%
    ggplot() +
        geom_point(aes(ROI, cells_per_mm2)) + 
        theme_minimal(base_size = 15)  + 
        theme(axis.text.x = element_text(angle = 90, hjust = 1, size = 8)) +
        ylab(&quot;Cells per mm2&quot;) + xlab(&quot;&quot;)</code></pre>
<p><img src="figure/1_QC_mask.Rmd/cells per mm2-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="differences-in-staining-patterns" class="section level3">
<h3>Differences in staining patterns</h3>
<pre class="r"><code>#pdf(&quot;allchannels_multidimplot.pdf&quot;, width = 15, height = 60)
multi_dittoPlot(spe, vars = use_channel,
               group.by = &quot;ROI&quot;, 
               plots = c(&quot;ridgeplot&quot;), 
               assay = &quot;exprs&quot;)</code></pre>
<pre><code>Picking joint bandwidth of 0.0548</code></pre>
<pre><code>Picking joint bandwidth of 0.0303</code></pre>
<pre><code>Picking joint bandwidth of 0.00502</code></pre>
<pre><code>Picking joint bandwidth of 0.0126</code></pre>
<pre><code>Picking joint bandwidth of 0.0122</code></pre>
<pre><code>Picking joint bandwidth of 0.121</code></pre>
<pre><code>Picking joint bandwidth of 0.0114</code></pre>
<pre><code>Picking joint bandwidth of 0.112</code></pre>
<pre><code>Picking joint bandwidth of 0.0117</code></pre>
<pre><code>Picking joint bandwidth of 0.0162</code></pre>
<pre><code>Picking joint bandwidth of 0.00864</code></pre>
<pre><code>Picking joint bandwidth of 0.0164</code></pre>
<pre><code>Picking joint bandwidth of 0.0308</code></pre>
<pre><code>Picking joint bandwidth of 0.0244</code></pre>
<pre><code>Picking joint bandwidth of 0.0601</code></pre>
<pre><code>Picking joint bandwidth of 0.101</code></pre>
<pre><code>Picking joint bandwidth of 0.0752</code></pre>
<pre><code>Picking joint bandwidth of 0.123</code></pre>
<pre><code>Picking joint bandwidth of 0.0579</code></pre>
<pre><code>Picking joint bandwidth of 0.058</code></pre>
<p><img src="figure/1_QC_mask.Rmd/staining QC-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()</code></pre>
</div>
<div id="umap-and-tsne-plots" class="section level3">
<h3>UMAP and TSNE plots</h3>
</div>
<div id="assessment-of-potential-batch-effects" class="section level3">
<h3>Assessment of potential “batch effects”</h3>
<p>#Umap with targets overlaid</p>
</div>
<div id="save-data" class="section level3">
<h3>Save data</h3>
<pre class="r"><code>saveRDS(spe, file.path(&quot;output&quot;, &quot;spe1.rds&quot;))</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_Switzerland.utf8  LC_CTYPE=English_Switzerland.utf8   
[3] LC_MONETARY=English_Switzerland.utf8 LC_NUMERIC=C                        
[5] LC_TIME=English_Switzerland.utf8    

attached base packages:
[1] stats4    stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] gtools_3.9.3                cowplot_1.1.1              
 [3] sp_1.5-0                    SeuratObject_4.1.1         
 [5] Seurat_4.1.1                harmony_0.1.0              
 [7] Rcpp_1.0.9                  batchelor_1.12.3           
 [9] patchwork_1.1.2             scater_1.24.0              
[11] mclust_5.4.10               scuttle_1.6.2              
[13] ggrepel_0.9.1               forcats_0.5.2              
[15] dplyr_1.0.9                 purrr_0.3.4                
[17] readr_2.1.2                 tidyr_1.2.0                
[19] tibble_3.1.7                tidyverse_1.3.2            
[21] viridis_0.6.2               viridisLite_0.4.1          
[23] RColorBrewer_1.1-3          dittoSeq_1.8.1             
[25] ggplot2_3.3.6               stringr_1.4.1              
[27] openxlsx_4.2.5              cytomapper_1.8.0           
[29] EBImage_4.38.0              imcRtools_1.3.7            
[31] SpatialExperiment_1.6.0     SingleCellExperiment_1.18.0
[33] SummarizedExperiment_1.26.1 Biobase_2.56.0             
[35] GenomicRanges_1.48.0        GenomeInfoDb_1.32.2        
[37] IRanges_2.30.0              S4Vectors_0.34.0           
[39] BiocGenerics_0.42.0         MatrixGenerics_1.8.1       
[41] matrixStats_0.62.0          workflowr_1.7.0            

loaded via a namespace (and not attached):
  [1] scattermore_0.8           R.methodsS3_1.8.2        
  [3] bit64_4.0.5               knitr_1.40               
  [5] irlba_2.3.5               DelayedArray_0.22.0      
  [7] R.utils_2.12.0            rpart_4.1.16             
  [9] data.table_1.14.2         RCurl_1.98-1.7           
 [11] generics_0.1.3            ScaledMatrix_1.4.0       
 [13] callr_3.7.2               terra_1.5-34             
 [15] RANN_2.6.1                proxy_0.4-27             
 [17] future_1.28.0             bit_4.0.4                
 [19] tzdb_0.3.0                spatstat.data_2.2-0      
 [21] xml2_1.3.3                lubridate_1.8.0          
 [23] httpuv_1.6.5              assertthat_0.2.1         
 [25] gargle_1.2.1              xfun_0.31                
 [27] hms_1.1.2                 jquerylib_0.1.4          
 [29] evaluate_0.16             promises_1.2.0.1         
 [31] fansi_1.0.3               dbplyr_2.2.1             
 [33] readxl_1.4.1              igraph_1.3.4             
 [35] DBI_1.1.3                 htmlwidgets_1.5.4        
 [37] spatstat.geom_2.4-0       googledrive_2.0.0        
 [39] ellipsis_0.3.2            backports_1.4.1          
 [41] deldir_1.0-6              svgPanZoom_0.3.4         
 [43] sparseMatrixStats_1.8.0   vctrs_0.4.1              
 [45] ROCR_1.0-11               abind_1.4-5              
 [47] cachem_1.0.6              withr_2.5.0              
 [49] ggforce_0.3.3             progressr_0.11.0         
 [51] vroom_1.5.7               sctransform_0.3.4        
 [53] goftest_1.2-3             cluster_2.1.3            
 [55] svglite_2.1.0             lazyeval_0.2.2           
 [57] crayon_1.5.1              labeling_0.4.2           
 [59] edgeR_3.38.1              pkgconfig_2.0.3          
 [61] units_0.8-0               tweenr_1.0.2             
 [63] nlme_3.1-158              vipor_0.4.5              
 [65] rlang_1.0.4               globals_0.16.1           
 [67] miniUI_0.1.1.1            lifecycle_1.0.1          
 [69] modelr_0.1.9              rsvd_1.0.5               
 [71] cellranger_1.1.0          rprojroot_2.0.3          
 [73] polyclip_1.10-0           lmtest_0.9-40            
 [75] tiff_0.1-11               Matrix_1.4-1             
 [77] raster_3.5-21             zoo_1.8-10               
 [79] Rhdf5lib_1.18.2           reprex_2.0.2             
 [81] beeswarm_0.4.0            RTriangle_1.6-0.10       
 [83] whisker_0.4               ggridges_0.5.3           
 [85] processx_3.7.0            googlesheets4_1.0.1      
 [87] pheatmap_1.0.12           png_0.1-7                
 [89] rjson_0.2.21              bitops_1.0-7             
 [91] shinydashboard_0.7.2      getPass_0.2-2            
 [93] R.oo_1.25.0               KernSmooth_2.23-20       
 [95] rhdf5filters_1.8.0        DelayedMatrixStats_1.18.0
 [97] classInt_0.4-7            spatstat.random_2.2-0    
 [99] parallelly_1.32.1         jpeg_0.1-9               
[101] beachmat_2.12.0           scales_1.2.1             
[103] ica_1.0-3                 magrittr_2.0.3           
[105] plyr_1.8.7                zlibbioc_1.42.0          
[107] compiler_4.2.1            dqrng_0.3.0              
[109] concaveman_1.1.0          fitdistrplus_1.1-8       
[111] cli_3.3.0                 XVector_0.36.0           
[113] listenv_0.8.0             pbapply_1.5-0            
[115] ps_1.7.1                  mgcv_1.8-40              
[117] MASS_7.3-58               tidyselect_1.1.2         
[119] stringi_1.7.8             highr_0.9                
[121] yaml_2.3.5                BiocSingular_1.12.0      
[123] locfit_1.5-9.6            grid_4.2.1               
[125] sass_0.4.2                tools_4.2.1              
[127] future.apply_1.9.1        parallel_4.2.1           
[129] rstudioapi_0.14           git2r_0.30.1             
[131] gridExtra_2.3             farver_2.1.1             
[133] Rtsne_0.16                ggraph_2.0.5             
[135] DropletUtils_1.16.0       rgeos_0.5-9              
[137] digest_0.6.29             shiny_1.7.2              
[139] broom_1.0.1               later_1.3.0              
[141] RcppAnnoy_0.0.19          httr_1.4.4               
[143] sf_1.0-8                  distances_0.1.8          
[145] colorspace_2.0-3          tensor_1.5               
[147] reticulate_1.26           rvest_1.0.3              
[149] fs_1.5.2                  splines_4.2.1            
[151] uwot_0.1.14               spatstat.utils_2.3-1     
[153] graphlayouts_0.8.0        plotly_4.10.0            
[155] systemfonts_1.0.4         xtable_1.8-4             
[157] jsonlite_1.8.0            tidygraph_1.2.1          
[159] R6_2.5.1                  pillar_1.8.1             
[161] htmltools_0.5.3           mime_0.12                
[163] nnls_1.4                  glue_1.6.2               
[165] fastmap_1.1.0             DT_0.24                  
[167] BiocParallel_1.30.3       BiocNeighbors_1.14.0     
[169] fftwtools_0.9-11          class_7.3-20             
[171] codetools_0.2-18          utf8_1.2.2               
[173] spatstat.sparse_2.1-1     lattice_0.20-45          
[175] bslib_0.4.0               ResidualMatrix_1.6.1     
[177] ggbeeswarm_0.6.0          leiden_0.4.2             
[179] magick_2.7.3              zip_2.2.1                
[181] survival_3.3-1            limma_3.52.2             
[183] rmarkdown_2.16            munsell_0.5.0            
[185] e1071_1.7-11              rhdf5_2.40.0             
[187] GenomeInfoDbData_1.2.8    HDF5Array_1.24.1         
[189] reshape2_1.4.4            haven_2.5.1              
[191] gtable_0.3.1              spatstat.core_2.4-4      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
